datarepo-clienttests/src/main/java/scripts/uploadscripts/SummariesToBigQueryJade.java

```diff
@@ -0,0 +1,240 @@
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import runner.TestRunner;
+import runner.TestScriptResult;
+import runner.config.TestConfiguration;
+import runner.config.TestScriptSpecification;
+import uploader.UploadScript;
+
+public class SummariesToBigQueryJade extends UploadScript {
+  private static final Logger logger = LoggerFactory.getLogger(SummariesToBigQueryJade.class);
+
+  /** Public constructor so that this class can be instantiated via reflection. */
+  public SummariesToBigQueryJade() {}
+
+  protected String projectId; // google project id
+  protected String datasetName; // big query dataset name
+
+  protected TestConfiguration renderedTestConfiguration;
+  protected TestRunner.TestRunSummary testRunSummary;
+  protected MeasurementCollectionScript.MeasurementResultSummary[] measurementCollectionSummaries;
+
+  /**
+   * Setter for any parameters required by the upload script. These parameters will be set by the
+   * Result Uploader based on the current Upload List, and can be used by the upload script methods.
+   *
+   * @param parameters list of string parameters supplied by the upload list
+   */
+  public void setParameters(List<String> parameters) throws Exception {
+    if (parameters == null || parameters.size() < 2) {
+      throw new IllegalArgumentException(
+          "Must provide BigQuery project_id and dataset_name in the parameters list");
+    }
+    projectId = parameters.get(0);
+    datasetName = parameters.get(1);
+  }
+
+  private static String testRunTableName = "testRun";
+  private static String testScriptResultsTableName = "testScriptResults";
+  private static String measurementCollectionTableName = "measurementCollection";
+
+  /**
+   * Upload the test results saved to the given directory. Results may include Test Runner
+   * client-side output and any relevant measurements collected.
+   */
+  public void uploadResults(Path outputDirectory) throws Exception {
+    // get a BigQuery client object
+    logger.debug("BigQuery project_id:dataset_name: {}:{}", projectId, datasetName);
+    BigQuery bigQueryClient = BigQueryUtils.getClient(projectId);
+
+    // read in TestConfiguration TestRunSummary, and array of
+    // MeasurementCollectionScript.MeasurementResultSummary objects
+    renderedTestConfiguration = TestRunner.getRenderedTestConfiguration(outputDirectory);
+    testRunSummary = TestRunner.getTestRunSummary(outputDirectory);
+    measurementCollectionSummaries =
+        MeasurementCollector.getMeasurementCollectionSummaries(outputDirectory);
+
+    // insert a single row into testRun
+    if (BigQueryUtils.checkRowExists(
+        bigQueryClient, projectId, datasetName, testRunTableName, "id", testRunSummary.id)) {
+      logger.info(
+          "A row with this id already exists in the "
+              + testRunTableName
+              + " table. Inserting a duplicate.");
+    }
```

Comment:
If we're planning on inserting the row in either case, do we need to perform this check?

Reply:
No, we don't need to. But I think it's helpful for debugging. In general, I don't think we'll ever be inserting something twice and I want to understand what happened if we do see duplicate rows. I considered throwing an error here instead, but figured it's always better to store results and figure out how to deal with duplicates after the fact.